File: src/main/java/com/github/nsc/de/compiler/parser/node/variables/VariableIncreaseNode.java
package com.github.nsc.de.compiler.parser.node.variables;

import com.github.nsc.de.compiler.parser.node.ValuedNode;

public class VariableIncreaseNode implements ValuedNode {

    private final ValuedNode variable;

    public VariableIncreaseNode(ValuedNode variable) {
        this.variable = variable;
    }

    public ValuedNode getVariable() {
        return variable;
    }

    @Override
    public String toString() {
        return '{' + this.getVariable().toString() + "++}";
    }
}

File: parser/src/commonMain/kotlin/io/github/shakelang/shake/parser/node/variables/VariableIncreaseNode.kt
package io.github.shakelang.shake.parser.node.variables

import io.github.shakelang.shake.parser.node.ValuedNode
import io.github.shakelang.parseutils.characters.position.PositionMap

class VariableIncreaseNode(map: PositionMap, val variable: ValuedNode, val operatorPosition: Int) : ValuedNode(map) {

    override fun toJson(): Map<String, *> = mapOf("name" to "VariableModAssignmentNode", "variable" to variable.json)

}